CreateRCConfiguration/handler.ts

```diff
@@ -0,0 +1,184 @@
+  key: detailsAuth.key,
+  type: detailsAuth.type
+});
+
+/**
+ * Convert the prod environment from snake case to camel case
+ * */
+export const getModelProdEnvironment = (
+  prodEnvironment: RCConfigurationProdEnvironment
+): RCEnvironmentConfig => ({
+  baseUrl: prodEnvironment.base_url,
+  detailsAuthentication: getModelDetailsAuthentication(
+    prodEnvironment.details_authentication
+  )
+});
+
+/**
+ * Convert the test environment from snake case to camel case
+ * */
+const getModelTestEnvironment = (
+  testEnv: RCConfigurationTestEnvironment
+): RCTestEnvironmentConfig => ({
+  baseUrl: testEnv.base_url,
+  detailsAuthentication: getModelDetailsAuthentication(
+    testEnv.details_authentication
+  ),
+  testUsers: testEnv.test_users
+});
+
+/**
+ * Convert the RCConfiguration from snake case to camel case
+ * */
+export const makeNewRCConfigurationWithConfigurationId = (
+  generateConfigurationId: () => Ulid,
+  userId: NonEmptyString,
+  publicConfiguration: NewRCConfigurationPublic
+): RCConfiguration => ({
+  configurationId: generateConfigurationId(),
+  description: publicConfiguration.description,
+  disableLollipopFor: publicConfiguration.disable_lollipop_for,
+  hasPrecondition: publicConfiguration.has_precondition,
+  isLollipopEnabled: publicConfiguration.is_lollipop_enabled,
+  name: publicConfiguration.name,
+  prodEnvironment: publicConfiguration.prod_environment
+    ? getModelProdEnvironment(publicConfiguration.prod_environment)
+    : undefined,
+  testEnvironment: publicConfiguration.test_environment
+    ? getModelTestEnvironment(publicConfiguration.test_environment)
+    : undefined,
+  userId
+});
+
+interface IHandlerParameter {
+  readonly newRCConfiguration: NewRCConfigurationPublic;
+  readonly userId: NonEmptyString;
+}
+
+interface ICreateRCConfigurationHandlerParameter {
+  readonly rccModel: RCConfigurationModel;
+  readonly generateConfigurationId: () => Ulid;
+}
+
+type CreateRCConfigurationHandlerReturnType = (
+  parameter: IHandlerParameter
+) => Promise<
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  | IResponseSuccessRedirectToResource<RCConfigurationPublic, {}>
+  | IResponseErrorInternal
+>;
+
+type CreateRCConfigurationHandler = (
+  parameter: ICreateRCConfigurationHandlerParameter
+) => CreateRCConfigurationHandlerReturnType;
+
+export const createRCConfigurationHandler: CreateRCConfigurationHandler = ({
+  rccModel,
+  generateConfigurationId
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+}) => ({ newRCConfiguration, userId }) =>
+  pipe(
+    rccModel.create(
+      makeNewRCConfigurationWithConfigurationId(
+        generateConfigurationId,
+        userId,
+        newRCConfiguration
+      )
+    ),
+    TE.map(
+      flow(retrievedRCConfigurationToPublic, publicConfiguration =>
+        ResponseSuccessRedirectToResource(
+          publicConfiguration,
+          `/api/v1/remote-contents/configurations/`,
```

Comment:
Why redirecting?

Reply:
Actually it is not a redirect, this function just set the header `Location`.
The same thing was made with the `CreateMessage` so I used it here for consistency